### ISSUE TITLE
cTivo Default Format creates files unreadable by MPEG Streamclip

### ISSUE THREAD

**Issue by aofrailey** [dscottbuch/cTiVo#236]
Since upgrading to cTivo 2.5.1,  the app  I use to edit commercials from downloaded files (MPEG Streamclip) no longer recognizes mp4 files generated by cTivo.  MPEG Streamclip continues to work perfectly well on the files generated by  cTivo 2.4.4. 

How do I set up cTivo 2.5.1 such that it produces the same info/output  as did cTivo 2.4.4?

Enclosed is a file that shows:
1.  The error messages I get when attempting to open a cTivo 2.5.1 - generated mp4 file with MPEG Streamclip.
2. Screen shots of  "Get Info"  for files  generated by cTivo  2.4.4 versus cTivo 2.5.1.
[2017-02-19 cTivo 2.5.1 Issue.docx](https://github.com/dscottbuch/cTiVo/files/786319/2017-02-19.cTivo.2.5.1.Issue.docx)

**Comment by mackworth** [comment]
So, which Format are you using in 2.5.1?  And which in 2.4.4?  And are you using the commercial feature in ctivo, or manually finding the commercials?

You should probably use either mediainfo or subler to tell you exactly what's inside each video file.  Subler has a better UI, but mediainfo gives more details.

**Comment by aofrailey** [comment]
I used the Default in 2.5.1. I  no longer have 2.4.4.  I vaguely recall using "Apple TV". Could it have been that?

**Comment by aofrailey** [comment]
I manually find the commercials. My Tivo is supplied by my provider, Suddenlink. I have not had good luck with the "Mark" or "Skip" options.

**Comment by aofrailey** [comment]
I think Suddenlink's version of Tivo does not have pre-identified commercials.

**Comment by mackworth** [comment]
AppleTV is quite possible for version 2.4.4.  

BTW, you might try the built-in mark and Skip again with 2.5.1. The underlying program is greatly improved.   FYI, it doesn't use TiVos pre-identified feature; it checks for itself.

That being said, I show the same thing; MPEG streamclip doesn't work with Default MP4 files. I should note that Streamclip is getting quite old; it hasn't been updated in five years.

After further testing, the specific problem is that Streamclip doesn't let you have an AC3 audio stream in the file. A new feature with 2.5.1 is the option to put both the AC3 multichannel and the AAC stereo channel in the same file. If you don't care about this, there's a simple fix: use a Format that doesn't include that AC3 channel, such as the HB ones. 

Alternatively, you can import this [Default, no AC3 Format file](https://github.com/dscottbuch/cTiVo/files/788261/Default.no.AC3.plist.zip). Just download it, make sure it decompresses to `Default No AC3.plist` and then use the `File>Import Formats` command to import it and select `Default, no AC3` from your Format menu.  Now future downloads should work fine with Streamclip.  

For people who want to create a `no AC3` version of an existing AC3-generating Format (which use the ffmpeg_ed_ac3.sh script), the steps are simple:
1. Edit>Edit Formats
2. Select the built-in Format you want to base your AC3 on (e.g. 1080p); make sure it says `ffmpeg_edl_ac3.sh` for Encoder to use.
3. Duplicate button
4. Edit the Name field (e.g. `1080p, no AC3`)  
5. Add `-noAC3` to the Audio Options field
6. Save button

If you want to remove the AC3 streams from existing downloads that have it, you'll have to use the Terminal program.  Just run it and it will open a window. 

1. Copy/paste the following: `/Applications/cTiVo.app/Contents/MacOS/ffmpeg -i  ` (Add a space at the end)
2. Then drag your video file into the window, and it will add the filename to your command line.
3.  Copy/paste the following: `-vcodec copy -acodec copy -map 0:v:0 -map 0:a:0   ` (Again with a space at the end)
4. Now you need to give it the output file name, which has to be different.  One way to do this is to drag your video file into the window again, and move the cursor back to edit to a slightly different filename (e.g. from `OLDNAME.mp4` to `OLDNAME-1.mp4`). It will warn you if you don't change the name.
5. Then hit return, and it should copy the old file to the new filename, but without the AC3.  
6. Check the new file with Streamclip and then delete the old one.

Hope this works for you. Let me know!

**Comment by aofrailey** [comment]
The "Default, no AC3 Format File"  worked great!  Thank you so much for describing the process on how to do it.  I tried other editors (QuickTime and iMovie) but the editing process was so cumbersome, I almost decided to leave the commercials in.  MPEG Streamclip is the easiest to use of the three I tried. And, by the way, unlike most technical explanations,  which are so poorly worded as to be  unexecutable, your descriptions are coherent, clear and articulate.  

"Mark" and "Skip" - I executed the cTivo download once with "Mark" selected and once with "Skip" selected.  "Skip" worked fine, except it cut out some clips I want to retain, such as the portion of the show that identifies the major cast members at the beginning.  
I am unable to tell how "Mark" worked in MPEG Streamclip.  As far as I can see, MPEG Streamclip is unable to use the marked version because MPEG Streamclip does not recognize multiple clips or chapters.  

Bottom line is I probably will continue to manually cut out commercials.

Thank you again!

**Comment by mackworth** [comment]
Good to hear, and thanks for compliment!
